### ISSUE TITLE
Update `component--Display` definitions

### ISSUE THREAD

**Issue by mattnolting** [patternfly/patternfly#7033]
TLDR:

Redefining `display` props with CSS Variables causes [forced reflows](https://web.dev/articles/avoid-large-complex-layouts-and-layout-thrashing). 

**Dependency Chains:** If variables depend on other variables, it can create a complex web of dependencies. Changing one variable might have unintended consequences elsewhere in the stylesheet.

Example:

**Initial variable definition**

```
:root, .pf-v6-tabs {
  // Tabs List
  --#{$tabs}__list--Display: flex;
  ...
}
```

**Chained dependency**

```
.#{$tabs} {
  ...

  // Vertical
  &.pf-m-vertical {
    @each $breakpoint, $breakpoint-value in $pf-v6-c-tabs--breakpoint-map {
      $breakpoint-name: if($breakpoint != "base", -on-#{$breakpoint}, "");

      @include pf-v6-apply-breakpoint($breakpoint) {
        &.pf-m-expandable#{$breakpoint-name} {
          --#{$tabs}__list--Display: none; // this causes a forced reflow
          --#{$tabs}__toggle--Display: flex; // this causes a forced reflow
        }
      }
    }
  }
}
```

## Themeable variables vs `prop: val` declarations

Instead of using variables that we do intend to be modified in an example such as `.pf-v6-tabs`, updating to `prop: value` declarations would result in the same composition without causing `forced-reflows`. The updates should be configured as such. 

**Initial variable definition:** Remove the display property altogether and update to `prop: val`

Remove the variable

```
:root, .pf-v6-tabs {
  // Tabs List
  // --#{$tabs}__list--Display: flex;
  ...
}
```

Use the `prop` and `value` instead

```
.#{$tabs} {
  display: flex;
  ...

  // Vertical
  &.pf-m-vertical {
    @each $breakpoint, $breakpoint-value in $pf-v6-c-tabs--breakpoint-map {
      $breakpoint-name: if($breakpoint != "base", -on-#{$breakpoint}, "");

      @include pf-v6-apply-breakpoint($breakpoint) {
        &.pf-m-expandable#{$breakpoint-name} {
          .#{$tabs}__list {
            display: none;
          }

          .#{$tabs}__toggle {
            display: flex;
          }
        }
      }
    }
  }
}
```

Living doc: https://docs.google.com/document/d/1jVDQ5d2X5a7PNVWlKYqvZZwDZcwGVPXMusYbQnhHcME/edit#heading=h.l2cuovol0fpy
